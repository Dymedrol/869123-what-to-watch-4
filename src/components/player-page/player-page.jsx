import React, {PureComponent} from "react";
import PropTypes from "prop-types";

import {videoPlayerModes} from '../../const.js';
import withVideoPlayer from '../../hocs/withVideoPlayer/withVideoPlayer.jsx';
import {Player} from '../player/player.jsx';

const PlayerWrapper = withVideoPlayer(Player);

class PlayerPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      allMovies,
    } = this.props;

    const getCurentMovie = (movies, movieId) => {
      return movies.find((movie) => movie.id === movieId);
    };

    const movieId = parseInt(this.props.match.params.id, 10);
    const movie = getCurentMovie(allMovies, movieId);

    if (!movie) {
      return <h2>Loading...</h2>;
    }

    return (
      <PlayerWrapper movie={movie} isMuted={true} videoMode={videoPlayerModes.FULLSCREEN}/>
    );

  }

  onTabClickHandler(tab) {
    this.setState({
      activeTab: tab,
    });
  }
}

PlayerPage.propTypes = {
  allMovies: PropTypes.array.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  }),
};

export {PlayerPage};
