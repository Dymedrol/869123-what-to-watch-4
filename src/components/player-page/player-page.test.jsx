import React from "react";
import renderer from "react-test-renderer";
import {PlayerPage} from "./player-page.jsx";
import {BrowserRouter} from 'react-router-dom';

import MOVIES from "../../mocks/films.js";

it(`Проверка снепшота компонента PlayerPage`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <PlayerPage
            allMovies={MOVIES}
            match={{params: {id: `0`}, isExact: true, path: ``, url: ``}}
            playVideo={() => {}}
          />
        </BrowserRouter>, {
          createNodeMock: () => {
            return {};
          }})
    .toJSON();

  expect(tree).toMatchSnapshot();
});
