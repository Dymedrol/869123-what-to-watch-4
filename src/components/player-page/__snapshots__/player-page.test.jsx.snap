// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка снепшота компонента PlayerPage 1`] = `
<h2>
  Loading...
</h2>
`;
